h2/src/main/org/h2/table/Column.java

```diff
@@ -68,6 +68,7 @@
     private int displaySize;
     private Table table;
     private String name;
+    private int columnPos;
```

Comment:
Let me try to explain this.
In ignite part, we're giving an user a possibility to choose the order in which 'Key', 'Val' fields appear in the QueryEntity.fields and effectively in the list of table columns. That alone makes our implementation of Row descendants less effective because we cannot have the KEY_COL and VAL_COL be static 0 and 1 anymore and we're required to have the information about these columns indexes provided to each Row we create.

The introduction of columnPos allows to remap columnId of Key and Val to still be 0 and 1 and at the same time respect the user's desire to put Key/Val anywhere on the list of columns.